### PR TITLE
META-3158 comments addressed

### PR DESCRIPTION
**Pull request by BurakAltas**
## Change description

> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java [hunk 1]
```diff
@@ -149,7 +151,13 @@ public void deleteEntities(Collection<AtlasVertex> instanceVertices) throws Atla
...
 
             if (DEFERRED_ACTION_ENABLED) {
-                createClassificationOnlyPropagationDeleteTasksAndQueue(getAllClassificationEdges(deletionCandidateVertex), RequestContext.get().getDeletedEdgesIds());
+                Set<String> deletedEdgeIds = RequestContext.get().getDeletedEdgesIds();
+                Set<AtlasVertex> classificationVertices = new HashSet<>();
+                for (String deletedEdgeId : deletedEdgeIds) {
+                    AtlasEdge edge = graph.getEdge(deletedEdgeId);
+                    classificationVertices.addAll(GraphHelper.getPropagatableClassifications(edge));
+                }
+                createClassificationOnlyPropagationDeleteTasksAndQueue(classificationVertices, deletedEdgeIds);
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:160]: There is a problem here, here we are creating task for each edge and all classification which is very wrong. We need to create task for each edge and classification attached with it. What I mean by that:
In this implementation:
We have edge 1 with classification A,B attached, edge 2 with classification C,D attached. If we follow this approach we will create task for (1,A),(1,B),(1,C),(1,D)(2,A),(2,B),(2,C),(2,D)
But we want to create task for : (1,A),(1,B),(2,C),(2,D)

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java [hunk 2]
```diff
@@ -74,8 +74,9 @@ protected void run(Map<String, Object> parameters) throws AtlasBaseException {
                 entityGraphMapper.deleteClassificationOnlyPropagation(deletedEdgeIds);
             } else {
                 String deletedEdgeId          =  (String) parameters.get(PARAM_DELETED_EDGE_ID);
+                String classificationVertexId =  (String) parameters.get(PARAM_CLASSIFICATION_VERTEX_ID);
 
-                entityGraphMapper.deleteClassificationOnlyPropagation(deletedEdgeId);
+                entityGraphMapper.deleteClassificationOnlyPropagation(deletedEdgeId, classificationVertexId);
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java:79]: It is tested on old tasks right?

**Review by sumandas0 — CHANGES_REQUESTED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java [hunk 3]
```diff
@@ -406,7 +406,13 @@ public EntityMutationResponse mapAttributesAndClassifications(EntityMutationCont
...
 
                     if (DEFERRED_ACTION_ENABLED) {
-                        deleteDelegate.getHandler().createClassificationOnlyPropagationDeleteTasksAndQueue(getAllClassificationEdges(vertex), reqContext.getDeletedEdgesIds());
+                        Set<String> deletedEdgeIds = reqContext.getDeletedEdgesIds();
+                        Set<AtlasVertex> classificationVertices = new HashSet<>();
+                        for (String deletedEdgeId : deletedEdgeIds) {
+                            AtlasEdge edge = graph.getEdge(deletedEdgeId);
+                            classificationVertices.addAll(GraphHelper.getPropagatableClassifications(edge));
+                        }
+                        deleteDelegate.getHandler().createClassificationOnlyPropagationDeleteTasksAndQueue(classificationVertices, deletedEdgeIds);
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:415]: Same case here

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java [hunk 4]
```diff
@@ -475,7 +481,13 @@ public EntityMutationResponse mapAttributesAndClassifications(EntityMutationCont
...
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:490]: Here also

**Review by sumandas0 — CHANGES_REQUESTED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java (hunk 1 above)
Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java]: you are correct. can you please check once more ?

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java (hunk 2 above)
Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java:79]: I do not have old task in my local setup currently. I will test that in staging

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java (hunk 3 above)
Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java]: addressed

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java (hunk 4 above)
Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java]: addressed

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java (hunk 2 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java:79]: Do one thing create task in master branch then stop Atlas in local then checkout to feature branch and then again run Atlas, so by that old task pattern will be executed in new way.

Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java:79]: MAke sure it works well in local first before even going into staging or beta

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java [hunk 5]
```diff
@@ -3495,62 +3503,60 @@ public void deleteClassificationOnlyPropagation(String deletedEdgeId) throws Atl
             return;
         }
```
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:3505]: Here please add one check if the classification still attached to that edge or not, what happens sometime classification might be deleted from the vertex attached to that edge by classification deletion or only propagation deletion of other edge, in that case we will not want to continue traversing. Adding that check could optimize some cases.

**Pull request closed without merging**